crypto/s2n_openssl_x509.c

```diff
@@ -17,10 +17,25 @@
 
 #include "api/s2n.h"
 
-int s2n_openssl_x509_stack_pop_free(STACK_OF(X509) **cert_chain)
+S2N_CLEANUP_RESULT s2n_openssl_x509_stack_pop_free(STACK_OF(X509) **cert_chain)
 {
-    if (*cert_chain != NULL) {
-        sk_X509_pop_free(*cert_chain, X509_free);
-    }
-    return S2N_SUCCESS;
+    RESULT_ENSURE_REF(*cert_chain);
```

Comment:
Yup. See my comment down below, but in recap: most things don't let you free things multiple times, and I'd be a little suspect of anything that called `free_thing` multiple times where maybe it's valid, maybe it isn't. I assumed that this would be a low risk change to make since
- it is an internal API
- we have sufficient test coverage to ensure this changes doesn't break anything